### PR TITLE
Add the support of Kotlin via kotlinRMiner.

### PR DESCRIPTION
**Pull request by onewhl**
- I added a `jar` with kotlinRMiner to /lib temporarily so we have the possibility to discuss the provided API and change it if we need it before publishing kotlinRMiner in the maven repository.

- I added several wrappers for entities we get from RefactoringMiner and kotlinRMiner -- CodeRange, LocationInfo, Refactoring, RefactoringType.
 
- I added fixPath() method in Utils temporarily because kotlinRMiner provides pathes that look like `/home/zarina/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2020.2/4fe93bb81525f2fa7a6f0fd7ba41c3b9cce9e8b6/ideaIC-2020.2/bin/src/main/kotlin/person/Person.kt`. Of course, my projects aren't located in` .gradle/caches/...`, that's why I added the method to fix pathes. 
I guess it's because of the `kotlinRMiner.jar` location, but I'm not sure.

- All handlers (`AddAttributeAnnotationHandler`, `ExtractClassHandler`, and so on) were changed in a similar way -- I just added a method for processing a refactoring we get from kotlinRMiner.

- The most important changes were made in `CommitMiner` class.

- **Several tests broke** without a visible reason, I should explore it deeper. It seems that the tests' configuration broke for some reason, the logs look weird.

### CODE REVIEW

File: src/main/java/org/jetbrains/research/refactorinsight/data/RefactoringEntry.java
Comment (vovak) [inline src/main/java/org/jetbrains/research/refactorinsight/data/RefactoringEntry.java:91]: We have a bit of duplicated code here and in other places (e.g. in `InfoFactory` and `Handler`), because of dealing with `Refactoring` classes from two different miners. These classes, however, look nearly identical. 
Could it be possible to instead use a common interface for Refactoring, implemented by wrappers for these two classes, and delegate method calls to concrete implementations in these wrappers? Looks like this could save a bit of duplicated code. I may be missing something, though.

**Review by vovak — COMMENTED** [review]
Thank you for this great work. It's huge.

Looks like quite some of the duplicated code could be eliminated by using a couple more wrapper classes.

Please consider a few comments I've made, and address known issues (e.g. failing tests) before we merge.

File: src/main/java/org/jetbrains/research/refactorinsight/data/RefactoringEntry.java
Comment (onewhl) [reply src/main/java/org/jetbrains/research/refactorinsight/data/RefactoringEntry.java:91]: We should push refactoring entities from each miner through because we need to process each refactoring type in a specific way. See Handler.specify() method implementations.

File: src/main/java/org/jetbrains/research/refactorinsight/processors/SingleCommitRefactoringTask.java
Comment (onewhl) [reply src/main/java/org/jetbrains/research/refactorinsight/processors/SingleCommitRefactoringTask.java]: Thanks, fixed!

File: src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java
Comment (onewhl) [reply src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java]: I agree, fixed!

File: src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java
Comment (onewhl) [reply src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java]: Fixed!

File: src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java
Comment (onewhl) [reply src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java]: Yes, thanks! Fixed.

**Comment by onewhl** [comment]
@vovak, thank you very much for the review! I've fixed most of the comments and fixed the broken tests.

**Review by vovak — APPROVED** [review]
Thanks for addressing the comments. It's OK to merge after the current problems are resolved.

**Review by vovak — APPROVED** [review]

**Pull request merged**
